src/xenia/base/math.h

```diff
@@ -139,6 +139,7 @@ static_assert(sizeof(unsigned int) == sizeof(uint32_t));
 static_assert(sizeof(unsigned long long) == sizeof(uint64_t));
 inline uint32_t bit_count(uint32_t v) { return __builtin_popcount(v); }
 inline uint32_t bit_count(uint64_t v) { return __builtin_popcountll(v); }
+inline uint32_t bit_count(size_t v) { return __builtin_popcountl(v); }
```

Comment:
Iirc when i was trying to compile on mac, it was only one case in the arena alloc

Reply:
That call can be changed to `xe::is_pow2`.

Reply:
That call can be changed to `xe::is_pow2`, though it returns true for 0 as well — it can be handled with an additional assertion or with `align = std::max(align, size_t(1))`.

The overall handling of the alignment of `Chunk` there is pretty fragile, by the way — assuming that `malloc` returns 16-byte-aligned memory, while it's true only on 64-bit x86 systems; on 32-bit x86, `malloc` only guarantees 8-byte alignment, so it's better to use an aligned malloc function provided by the platform explicitly. Also that assertion says that the number 16 is the alignment of `Chunk`, which suggests that `alignof` may be used there instead. However, `Chunk` requires the alignment of a pointer actually — and there's no `alignas` to overalign the type. But there's no clear requirement that the `Chunk` class itself in isolation must be aligned. However, it may still be nice to move that 16 to a `static constexpr` with a static assertion that it must be `>= alignof(Chunk)`.